web/wp-content/plugins/tmbi-adobe-dtm/js/adobe_dtm.js

```diff
@@ -0,0 +1,388 @@
+/**
+ * Created by jeysaravana on 2017-01-11.
+ */
+
+var qs = (function() {
+	query_string_url = window.location.search.substr(1).split('&');
+	if ( query_string_url === "" ) return {};
+	var query_strings = {};
+	for ( var i = 0; i < query_string_url.length; ++i ) {
+		var parameter = query_string_url[i].split('=', 2);
+		if ( parameter.length === 1 ) {
+			query_strings[parameter[0]] = "";
+		} else {
+			query_strings[parameter[0]] = decodeURIComponent( parameter[1].replace( /\+/g, " " ) );
+		}
+	}
+	return query_strings;
+})();
+
+//Nelio A/B Testing
+if ( qs['nabt'] !== undefined ) {
+	//TODO setup adobe variables.
+	//for original => nabt = 0
+	//for alternatice => nabt = 1
+	var test_name = 'tohdesign:a_newdesign';
+	if ( qs['nabt'] === '1' ) {
+		test_name = 'tohdesign:b_currentdesign';
+	}
+	digitalData.ab = digitalData.ab||{};
+	digitalData.ab.testing = test_name; //qs['nabt']
+}
+
+jQuery( document ).ready(function($) {
+
+	if ( qs['ehid'] !== undefined ) {
+		localStorage.newsletter 	= qs['ehid'];
+		digitalData.newsletter      = digitalData.newsletter||{};
+		digitalData.newsletter.ehid = qs['ehid'];
+	}
+	if ( typeof( localStorage.newsletter ) != 'undefined' && typeof qs['ehid'] == 'undefined' ){
+		digitalData.newsletter      = digitalData.newsletter||{};
+		digitalData.newsletter.ehid = localStorage.newsletter;
+	}
+
+	//RD Related changes
+	$("#more-from-category").find('a').each(function() {
+		replace_data_attr( $(this), 'main stage - list', 'more about' );
+	});
+	$('.menu-footer-social-container').find('a').each(function() {
+		replace_data_attr( $(this), 'follow us', 'footer' );
+	});
+
+	$('.page-unsubscribe .ninja-forms-form').submit(function() {
+		do_adobe_data_analytics( 'Save Preference', 'newsletter preferences', 'opt out page' );
+	});
+
+	$('.joke .read-more').on('click', function() {
+
+		var joke_block = $(this).closest('.joke');
+		if (typeof do_adobe_data_analytics === 'function') {
+			var name = joke_block.find('.entry-header .entry-title a').text();
+			do_adobe_data_analytics(name, 'content navigation', 'main content overlay');
+		}
+	});
+
+    //TOH Related Changes
+    var login_status = 'false';
+	var userData = {"profile" : { "profileInfo" : {}}}
+    if ( getCookie( 'ID' ) !== undefined ) {
+        login_id = getCookie( 'ID' );
+        login_status = 'true';
+        userData["profile"]["profileInfo"]["profileID"] = login_id;
+
+    }
+    userData["profile"]["profileInfo"]["profileStatus"] = login_status;
+    digitalData.user = userData;
+
+	//FHM Related Changes
+	$('#genesis-sidebar-primary').find('.menu-social-profiles-container a').each(function() {
+		var social_profiles = $(this).attr('data-analytics-metrics');
+		social_profiles = social_profiles.replace('footer', 'right rail');
+		$(this).attr('data-analytics-metrics', social_profiles);
+	});
+
+	var $body = $("body");
+	$body.on("click", "[data-analytics-metrics]", function (e) {
```

Comment:
This event listener and the next one are generic and provide the core integration with ADTM. This, plus `adobe_data_analytics` function and its helpers, are probably the only JS code that should be provided by the ADTM plugin, while the rest should be integrated from the theme or plugins providing the related functionality.